### PR TITLE
Add a flag for controlling topic autoconfiguration by the Kafka binder

### LINKED ISSUE #504: Add a flag to disable topic creation
`spring-cloud-stream-binder-kafka` creates a topic if it's not already there. It also overwrites the number of partition and replication factor of any existing topic if the config in the application differs from what the topic currently has. Deleting a topic has been a troublesome experience for us (using 0.8.2) as the topic gets marked for deletion, and kafka doesn't start up. Would be great if there is a flag that can be set to disable topic creation by any app in the kafka.

### PR DESCRIPTION
**Pull request by mbogoevici**
Fixes #504

- introduces a new binder setting `autoConfigureTopics` that allows the user to disable the automatic creation of topics by the binder

### CODE REVIEW

File: spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc [hunk 1]
```diff
@@ -839,6 +839,15 @@ Mutually exclusive with `offsetUpdateTimeWindow`.
 Default: `0`. 
 spring.cloud.stream.kafka.binder.requiredAcks::
   The number of required acks on the broker.
++
+Default: `1`.
+spring.cloud.stream.kafka.binder.autoConfigureTopics::
+  If set to `true`, the binder will create new topics or add new partitions to existing topics automatically to match the requirements of the producers and consumers.
+If set to `false`, the binder will rely on the topics being already configured.
+In the latter case, if the topics do not exist or the partition count is smaller than the expected partition count, the binder will fail to start.
+Of note, this setting is independent of the `auto.topic.create.enable` of the broker and it does not influence it.
```
Comment (mbogoevici) [inline spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: .. of the `auto.topic.create.enable` _setting_ of the broker

File: spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java [hunk 2]
```diff
@@ -445,6 +455,88 @@ public void testSyncProducerMetadata() throws Exception {
 		assertTrue("Kafka Sync Producer should have been enabled.", producerConfiguration.getProducerMetadata().isSync());
 		producerBinding.unbind();
 	}
+	
+	@Test
+	public void testAutoConfigureTopicsDisabledFailsIfTopicMissing() throws Exception {
```
Comment (garyrussell) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java]: Do we need a test for topic exists, but with not enough partitions?

Comment (mbogoevici) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java]: I'll add one.

**Comment by mbogoevici** [comment]
Pushed.

**Comment by garyrussell** [comment]
Squashed, rebased, merged as 8d5bf66c7f79802cccc0fdc6218c13ac517b695b.

**Pull request closed without merging**